internal/controllers/topology/cluster/structuredmerge/serversidepathhelper.go

```diff
@@ -135,6 +139,7 @@ func (h *serverSidePatchHelper) Patch(ctx context.Context) error {
 	log := ctrl.LoggerFrom(ctx)
 	log.V(5).Info("Patching object", "Intent", h.modified)
 
+	h.modified.SetUID(h.originalUID)
```

Comment:
I assume we don't have to set it in allowedPaths (like e.g. metadata.name) as we simply don't set it before running through the diff logic

Reply:
Exactly, especially we have to carry it over somewhere from the original (if not nil) to the modified, because modified is often computed.